src/wrapped.rs

```diff
@@ -0,0 +1,112 @@
+#![allow(missing_docs)]
+
+use {StringInterner, Symbol, Iter, Values};
+use std::collections::hash_map::RandomState;
+use std::hash::BuildHasher;
+use std::ops::Deref;
+
+#[derive(Copy, Clone, Debug)]
+pub struct PooledStr<'pool, Sym: Symbol + 'pool = usize, H: BuildHasher + 'pool = RandomState> {
+	pool: &'pool StringPool<Sym, H>,
+	sym: Sym,
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> Eq for PooledStr<'pool, Sym, H> {}
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> PartialEq<Self> for PooledStr<'pool, Sym, H> {
+	fn eq(&self, other: &Self) -> bool {
+		self.sym == other.sym && ::std::ptr::eq(self.pool, other.pool)
+	}
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> Deref for PooledStr<'pool, Sym, H> {
+	type Target = str;
+	fn deref(&self) -> &str {
+		// in the future, maybe use resolve_unchecked
+		PooledStr::resolve(self)
+	}
+}
+
+impl<'pool, Sym: Symbol + 'pool, H: BuildHasher + 'pool> PooledStr<'pool, Sym, H> {
+	pub fn resolve(this: &Self) -> &str {
```

Comment:
Just curious: Why do you use `this: &Self` instead of simply `&self` here?

Reply:
The purpose is to make `PooledStr::resolve` not a method. At the use site, go though the `Deref` impl instead of `PooledStr::resolve`. `std::rc:Rc` uses a similar pattern. The point is to avoid adding methods to what's supposed to masquerade as the wrapped type.